### PR TITLE
TransactionsToApprove - unset reference if transaction is deeper than depth

### PR DESCRIPTION
**Pull request by alon-e**

### CODE REVIEW

File: src/main/java/com/iota/iri/service/API.java [hunk 1]
```diff
@@ -531,8 +531,16 @@ public static void incEllapsedTime_getTxToApprove(long ellapsedTime) {
         Hash referenceHash = null;
         if(reference != null) {
             referenceHash = new Hash(reference);
-            if(!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+            if (!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+                //reference not found
                 referenceHash = null;
+            } else {
+                TransactionViewModel transactionViewModel = TransactionViewModel.fromHash(instance.tangle, referenceHash);
+                if (transactionViewModel.snapshotIndex() != 0
```
Comment (Awnedion) [inline src/main/java/com/iota/iri/service/API.java]: Does this != 0 check mean that a very large block of unconfirmed transactions can be used as a reference? Is that safe?

File: src/main/java/com/iota/iri/service/API.java [hunk 2]
```diff
@@ -531,8 +531,16 @@ public static void incEllapsedTime_getTxToApprove(long ellapsedTime) {
...
+                        && transactionViewModel.snapshotIndex() < instance.milestone.latestSolidSubtangleMilestoneIndex - depth) {
```
Comment (th0br0) [inline src/main/java/com/iota/iri/service/API.java]: Depth needs to be bounded else it can be abused.

Comment (Awnedion) [reply src/main/java/com/iota/iri/service/API.java]: Depth is already bounded.

**Comment by th0br0** [comment]
Closing because #501 contained these changes.

**Pull request closed without merging**